assets/js/dashboard/stats/graph/visitor-graph.js

```diff
@@ -335,45 +401,70 @@ export default class VisitorGraph extends React.Component {
   }
 
   fetchGraphData() {
-    if (this.state.metric) {
-      api.get(`/api/stats/${encodeURIComponent(this.props.site.domain)}/main-graph`, this.props.query, {metric: this.state.metric || 'none'})
+    if (!this.state.metric) {
+      this.setState({ loadingMainGraph: false, graphData: null })
+      return
+    }
+
+    const url = `/api/stats/${encodeURIComponent(this.props.site.domain)}/main-graph`
+    let params = {metric: this.state.metric || 'none'}
+    if (this.state.interval) { params.interval = this.state.interval }
+
+    api.get(url, this.props.query, params)
       .then((res) => {
-        this.setState((state) => ({ loading: state.loading-2, graphData: res }))
+        this.setState({ loadingMainGraph: false, graphData: res })
         return res
       })
-    } else {
-      this.setState((state) => ({ loading: state.loading-2, graphData: null }))
-    }
+      .catch((err) => {
+        console.log(err)
+        this.setState({ loadingMainGraph: false, graphData: false })
+      })
   }
 
   fetchTopStatData() {
     api.get(`/api/stats/${encodeURIComponent(this.props.site.domain)}/top-stats`, this.props.query)
       .then((res) => {
-        this.setState((state) => ({ loading: state.loading-1, topStatData: res }))
+        this.setState({ loadingTopStats: false, topStatData: res })
         return res
       })
-    }
+  }
 
   renderInner() {
     const { query, site } = this.props;
-    const { graphData, metric, topStatData, loading } = this.state;
+    const { graphData, metric, topStatData, loadingTopStats, loadingMainGraph } = this.state;
 
     const theme = document.querySelector('html').classList.contains('dark') || false
 
-    if ((loading <= 1 && topStatData) || (topStatData && graphData)) {
-      return (
-          <LineGraphWithRouter graphData={graphData} topStatData={topStatData} site={site} query={query} darkTheme={theme} metric={metric} updateMetric={this.updateMetric} />
-      )
-    }
+    const mainGraphReadyOrRefreshing = (!loadingMainGraph || loadingMainGraph === 2)
+    const noMetricOrLoaded = (!metric || loadingMainGraph === 2)
+    const topStatAndGraphLoaded = !!(topStatData && graphData)
+
+    const showGraph =
+      !loadingTopStats &&
+      mainGraphReadyOrRefreshing &&
+      (topStatData && noMetricOrLoaded || topStatAndGraphLoaded)
+
+    return (
+      <FadeIn show={showGraph}>
+        <LineGraphWithRouter graphData={graphData} topStatData={topStatData} site={site} query={query} darkTheme={theme} metric={metric} updateMetric={this.updateMetric} updateInterval={this.updateInterval}/>
+      </FadeIn>
+    )
   }
 
   render() {
-    const {metric, topStatData, graphData} = this.state
+    const {metric, loadingMainGraph, loadingTopStats} = this.state
+    const graphClassName = classNames({
```

Comment:
The convention that makes classes easier to read/scan is:
1. If class is not conditional, list it as a string in the first argument
2. If class is conditional, list it as object key in second argument

```suggestion
    const graphClassName = classNames('mx-auto loading', {
      'pt-52 sm:pt-56 md:pt-60': loadingMainGraph == 2,
      'pt-32 sm:pt-36 md:pt-48': loadingMainGraph !== 2 && metric,
      'pt-16 sm:pt-14 md:pt-18 lg:pt-5': loadingMainGraph !== 2 && !metric
    })
```